spec.md

```diff
@@ -1258,6 +1264,15 @@ can be mounted on all app instances simultaneously.
 }
 ```
 
+### Failure Cleanup
+
+After an unsuccessful bind attempt, a Service Broker MAY have stray resources
+persisted as a side effect. In order to prevent resource leaks, if a bind request
+has been sent to a Service Broker, then the Platform MUST send an unbind request
+at some point before cleaning up its internal representation of the Service
+Binding. Should that unbind attempt fail, the Platform SHOULD keep retrying until
+the Service Broker responds with a success.
```

Comment:
Do we want to say 'retry up to its internal limit' or something similar?  We mention limits of polling elsewhere.

One thing that would be useful is the ability to list service instances and bindings.  However, I suggest that that is such a radical change, and so unlikely to be implemented correctly (broker authors will have to implement acl filtering, for example), that we should consider 'list' and 'watch' in the API to be v3 things.

Reply:
I'm happy to add a mention of an internal limit. The original text didn't mention one in the orphan mitigation case, but I think it's a useful weakening, especially since polling already has the concept.